core/services/ardupilot_manager/firmware/FirmwareInstall.py

```diff
@@ -16,13 +16,355 @@
 from typedefs import FirmwareFormat, FlightController, Platform, PlatformType
 
 
-def get_board_id(platform: Platform) -> int:
+def get_board_id(board: Union[FlightController, str]) -> int:
+    print(board)
+    if isinstance(board, FlightController):
+        platform = board.name
+    else:
+        platform = board
     ardupilot_board_ids = {
         Platform.Pixhawk1: 9,
         Platform.Pixhawk4: 50,
         Platform.Pixhawk6X: 53,
         Platform.Pixhawk6C: 56,
         Platform.CubeOrange: 140,
+        "PX4_FMU_V1": 5,
```

Comment:
We should refactor typedefs.Platform no ? It would be nice to unify both implementation

Reply:
We can, yes. I was trying to not touch stuff that currently works.
We could leave this PR for after 1.4 and then I can rework it to unify everything.